include/proxsuite/proxqp/dense/wrapper.hpp

```diff
@@ -272,6 +273,11 @@ struct QP
               optional<T> mu_eq = nullopt,
               optional<T> mu_in = nullopt)
   {
+    if (!work.isInitialized) {
+      PROXSUITE_THROW_PRETTY(true,
+                             std::runtime_error,
+                             "init method needs to be called before update.")
+    }
```

Comment:
Could you call `init` in the back of the user and skip useless treatments in what follows?
It will simplify the life of the end-user.